pkg/agent/openflow/packetin.go

```diff
@@ -39,6 +39,45 @@ type PacketInHandler interface {
 }
 
 const (
+	// PacketIn categorys below are used to distribute packetIn to specific handler.
+	// PacketIn category should be loaded in the first byte of packet-in userdata.
+	// PacketInCategoryTF is used for traceflow.
+	PacketInCategoryTF ofpPacketInCategory = iota
+	// PacketInCategoryNP is used for packet-in messages related to Network Policy,
+	// including: Logging, Reject, Deny.
+	PacketInCategoryNP
+	// PacketInCategoryDNS is used for DNS response packet-in messages.
+	PacketInCategoryDNS
+	// PacketInCategoryIGMP is used for IGMP packet-in message.
+	PacketInCategoryIGMP
+	// PacketInCategorySvcReject is used to process the Service packet not matching any
+	// Endpoints within packet-in message.
+	PacketInCategorySvcReject
+
+	// PacketIn operations below are used to decide which operation(s) should be
+	// executed by handler. It(they) should be loaded in the second byte of the
+	// packet-in userdata. Operations for different handlers could share the value.
+	// If there is only one operation for a handler, then there is no need to provide a
+	// operation.
+	// For example, if a packet-in need Reject and Logging, the userdata of the
+	// packet-in will be []byte{1, 0b11}. The first byte indicate that this packet-in
+	// should be sent to NetworkPolicy packet-in handler(PacketInCategoryNP). And the
+	// second byte, which is 0b1 & 0b10 indicating that it need
+	// PacketInNPLoggingOperation and PacketInNPRejectOperation.
+	// PacketInNoMultiOperation is used when there is only one operation for a handler.
+	PacketInNoMultiOperation = 0b0
```

Comment:
Shall we remove it if all features construct the userdata value ?

Reply:
Removed it after changing to use `[]byte` for `SendToController`.